v2/swagger.yaml

```diff
@@ -0,0 +1,629 @@
+        items:
+          type: boolean
+      dependents_count:
+        type: array
+        items:
+          type: integer
+  Package:
+    title: Component Package Specific Details
+    description: Component package details
+    properties:
+      libio_dependents_repos:
+        type: array
+        items:
+          type: string
+      libio_latest_release:
+        type: array
+        items:
+          type: string
+      gh_prs_last_year_opened:
+        type: array
+        items:
+          type: integer
+      libio_latest_version:
+        type: array
+        items:
+          type: string
+      last_updated:
+        type: array
+        items:
+          type: number
+          format: float
+      gh_subscribers_count:
+        type: array
+        items:
+          type: number
+      gh_stargazers:
+        type: array
+        items:
+          type: number
+      libio_dependents_projects:
+        type: array
+        items:
+          type: string
+      gh_forks:
+        type: array
+        items:
+          type: integer
+      libio_usedby:
+        type: array
+        items:
+          type: string
+      gh_issues_last_month_closed:
+        type: array
+        items:
+          type: integer
+      ecosystem:
+        type: array
+        items:
+          type: string
+      gh_prs_last_year_closed:
+        type: array
+        items:
+          type: integer
+      gh_issues_last_year_opened:
+        type: array
+        items:
+          type: integer
+      gh_prs_last_month_closed:
+        type: array
+        items:
+          type: integer
+      gh_prs_last_month_opened:
+        type: array
+        items:
+          type: integer
+      vertex_label:
+        type: array
+        items:
+          type: string
+      gh_contributors_count:
+        type: array
+        items:
+          type: integer
+      gh_issues_last_month_opened:
+        type: array
+        items:
+          type: integer
+      gh_issues_last_year_closed:
+        type: array
+        items:
+          type: integer
+      latest_version:
+        type: array
+        items:
+          type: integer
+      gh_open_issues_count:
+        type: array
+        items:
+          type: integer
+      name:
+        type: array
+        items:
+          type: string
+      package_relative_used:
+        type: array
+        items:
+          type: string
+      package_dependents_count:
+        type: array
+        items:
+          type: integer
+      libio_total_releases:
+        type: array
+        items:
+          type: string
+      tokens:
+        type: array
+        items:
+          type: string
+  ComponentAnalysesDataDetails:
+    title: Component Analyses Package and Version
+    description: Component analyses data for package and version
+    properties:
+      version:
+        $ref: "#/definitions/Version"
+      package:
+        $ref: "#/definitions/Package"
+  PremiumVulnerabilityFields:
```

Comment:
Does this indicate private vulnerabilities ?

Reply:
How do we differentiate public and private vulnerabilities ?

Reply:
Dependency model has following 2 attributes to distinguish public and private vulnerabilities,
 
1. public_vulnerabilities
2. private_vulnerabilities

Both of the above fields are type of `Vulnerability` model. `Vulnerability` contains `PremiumVulnerabilityFields` which will have more information about a vulnerability for a registered user, for unregistered user it will be null.